internal/service/profile/training.go

```diff
@@ -0,0 +1,72 @@
+package profile
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/rs/zerolog/log"
+	openapi "github.com/ucl-arc-tre/portal/internal/openapi/web"
+	"github.com/ucl-arc-tre/portal/internal/service/profile/certificate"
+	"github.com/ucl-arc-tre/portal/internal/types"
+)
+
+func (s *Service) UpdateTraining(user types.User, data openapi.ProfileTrainingUpdate) (openapi.ProfileTrainingResponse, error) {
+	if data.CertficateContentPdfBase64 == nil {
+		log.Debug().Any("username", user.Username).Msg("Empty certificate content")
+		return openapi.ProfileTrainingResponse{CertificateIsValid: ptr(false)}, nil
+	}
+	switch data.Kind {
+	case openapi.TrainingKindNhsd:
+		return s.updateNHSD(user, data)
+	}
+	return openapi.ProfileTrainingResponse{}, fmt.Errorf("unsupported training kind")
+}
+
+func (s *Service) updateNHSD(
+	user types.User,
+	data openapi.ProfileTrainingUpdate,
+) (openapi.ProfileTrainingResponse, error) {
+	certificate, err := certificate.ParseNHSDCertificate(*data.CertficateContentPdfBase64)
+	response := openapi.ProfileTrainingResponse{
+		CertificateIsValid: ptr(false),
+	}
+	if err != nil {
+		response.CertificateMessage = ptr("Failed to parse certificate")
+		return response, err
+	}
+	if !certificate.HasIssuedAt() {
+		return response, fmt.Errorf("certificate must have an issused at time to be saved")
```

Comment:
question: should this be an error or just a non valid certificate? on reflection I'm thinking perhaps latter

Reply:
non valid + message to contact/automated contact (or some direction of this that we pick up on the front end and interpret). In the cases where it doesn't have the issue time, is it because the parser hasn't picked it up?